sql/catalyst/src/main/scala/org/apache/spark/sql/types/AbstractDataType.scala

```diff
@@ -34,9 +34,16 @@ private[sql] abstract class AbstractDataType {
   private[sql] def defaultConcreteType: DataType
 
   /**
-   * Returns true if this data type is a parent of the `childCandidate`.
+   * Returns true if this data type is the same type as `other`.  This is different that equality
+   * as equality will also consider data type parametrization, such as decimal precision.
```

Comment:
while you are at it, i think you can just give an example here, e.g.

DecimalType.isSameType(DecimalType(10, 2)) returns true, but NumericType.isSameType(...) returns false

to make it more clear.